src/controls/TextInput.js

```diff
@@ -16,17 +17,92 @@ function TextInput(theme, skinName) {
     // show and load background image as skin (exploiting skin states)
     this.skinName = skinName || TextInput.SKIN_NAME;
     this._validStates = this._validStates || InputControl.stateNames;
+    this._type = TextInput.TEXT;
+    this._acceptComma = false;
+    this.check = false;
 
     InputControl.call(this, theme);
 }
-
 TextInput.prototype = Object.create(InputControl.prototype);
 TextInput.prototype.constructor = TextInput;
+TextInput.uber = InputControl.prototype;
 module.exports = TextInput;
 
 // name of skin
 TextInput.SKIN_NAME = 'text_input';
 
+/**
+ * Text type: all input allowed
+ *
+ * @property TEXT
+ * @static
+ * @final
+ * @type String
+ */
+TextInput.TEXT = 'text';
+
+/**
+ * Number type: only numbers are accepted
+ *
+ * @property NUMBER
+ * @static
+ * @final
+ * @type String
+ */
+TextInput.NUMBER = 'number';
+
+/**
+ * names of possible types for a TextInput
+ *
+ * @property types
+ * @static
+ * @final
+ * @type String
+ */
+TextInput.types = [
+  TextInput.TEXT, TextInput.NUMBER
+];
+
+/*
+ * set type attribute
+ */
+Object.defineProperty(TextInput.prototype, 'type', {
+    get: function () {
+        return this._type;
+    },
+    set: function (type) {
+        if (TextInput.types.indexOf(type) === -1) {
+            return;
+        }
+        this._type = type;
+        if (type === TextInput.NUMBER) {
+            if (this.text && !this.validateTypeOfNumber(this.text)) {
+                this.text = '';
+            }
+            this.interceptors.push(this.validateTypeOfNumber);
```

Comment:
create `test/controls/TextInput.js` with content:

```javascript
describe("TextInput", function() {
  beforeEach(function(){
    // cleanup - make sure global theme is not set
    GOWN.Theme.removeTheme();
    new GOWN.TestTheme();
  });

  it("allow non numeric chars after changing type from number to text", function() {
    var input = new GOWN.TextInput();

    expect(input.type).equal(GOWN.TextInput.TEXT);

    input.type = GOWN.TextInput.NUMBER;
    expect(input.type).equal(GOWN.TextInput.NUMBER);
    input.type = GOWN.TextInput.TEXT;
    expect(input.type).equal(GOWN.TextInput.TEXT);

    input.insertChar("a");
    input.insertChar("1");
    input.insertChar(".");
    input.insertChar("5");
    input.insertChar("b");

    expect(input.text).equal("a1.5b");
  });
});
```

and to `test/index.js` add 
```javascript
require('./controls/TextInput');
```

Please make sure the unit test doesnt fail